cmd/whereabouts_test.go

```diff
@@ -531,6 +531,194 @@ var _ = Describe("Whereabouts operations", func() {
 
 	})
 
+	It("allocates an address using IPRanges notation", func() {
+		const ifname string = "eth0"
+		const nspath string = "/some/where"
+
+		backend := fmt.Sprintf(`"kubernetes": {"kubeconfig": "%s"}`, kubeConfigPath)
+		conf := fmt.Sprintf(`{
+			"cniVersion": "0.3.1",
+			"name": "mynet",
+			"type": "ipvlan",
+			"master": "foo0",
+			"ipam": {
+			  "type": "whereabouts",
+			  "log_file" : "/tmp/whereabouts.log",
+			  "log_level" : "debug",
+			  %s,
+			  "ipRanges": [{
+			    "range": "192.168.10.1/24"
+			  }]
+			}
+		}`, backend)
+
+		args := &skel.CmdArgs{
+			ContainerID: "dummy",
+			Netns:       nspath,
+			IfName:      ifname,
+			StdinData:   []byte(conf),
+			Args:        cniArgs(podNamespace, podName),
+		}
+
+		ipamConf, cniVersion, err := config.LoadIPAMConfig([]byte(conf), cniArgs(podNamespace, podName), "")
+		Expect(err).NotTo(HaveOccurred())
+		k8sClient = newK8sIPAM(
+			args.ContainerID,
+			ipamConf,
+			fakek8sclient.NewSimpleClientset(),
+			fake.NewSimpleClientset(
+				ipPool(ipamConf.IPRanges[0].Range, podNamespace)))
+
+		// Allocate the IP
+		r, raw, err := testutils.CmdAddWithArgs(args, func() error {
+			return cmdAdd(args, k8sClient, cniVersion)
+		})
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(strings.Index(string(raw), "\"version\":")).Should(BeNumerically(">", 0))
+
+		result, err := current.GetResult(r)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Gomega is cranky about slices with different caps
+
+		Expect(result.IPs).NotTo(BeEmpty())
+		Expect(result.IPs[0].Address).To(Equal(mustCIDR("192.168.10.1/24")))
+
+		// Release the IP
+		err = testutils.CmdDelWithArgs(args, func() error {
+			return cmdDel(args, k8sClient)
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("allocates DualStack address using IPRanges notation", func() {
+		const ifname string = "eth0"
+		const nspath string = "/some/where"
+
+		backend := fmt.Sprintf(`"kubernetes": {"kubeconfig": "%s"}`, kubeConfigPath)
+		conf := fmt.Sprintf(`{
+			"cniVersion": "0.3.1",
+			"name": "mynet",
+			"type": "ipvlan",
+			"master": "foo0",
+			"ipam": {
+			  "type": "whereabouts",
+			  "log_file" : "/tmp/whereabouts.log",
+			  "log_level" : "debug",
+			  %s,
+			  "ipRanges": [{
+			    "range": "192.168.10.1/24"
+			  }, {
+			    "range": "abcd::1/64"
+			  }]
+			}
+		}`, backend)
+
+		args := &skel.CmdArgs{
+			ContainerID: "dummy",
+			Netns:       nspath,
+			IfName:      ifname,
+			StdinData:   []byte(conf),
+			Args:        cniArgs(podNamespace, podName),
+		}
+
+		ipamConf, cniVersion, err := config.LoadIPAMConfig([]byte(conf), cniArgs(podNamespace, podName), "")
+		Expect(err).NotTo(HaveOccurred())
+		k8sClient = newK8sIPAM(
+			args.ContainerID,
+			ipamConf,
+			fakek8sclient.NewSimpleClientset(),
+			fake.NewSimpleClientset(
+				ipPool(ipamConf.IPRanges[0].Range, podNamespace)))
```

Comment:
Yes, you're absolutely correct. The other one is lazily created :)

Reply:
I think we should be consistent here ... either always lazy, or always create the pools.

Am I missing something ?

Reply:
As far as I remember, I didn't find any tests for lazy IP Pool creation, so though of initializing one lazily to test that.
But nevertheless, since it is out of scope of this PR (and also makes the test case confusing), I updated the code to create both the IP Pools :)
Thanks for pointing it out ^^